### PR TITLE
Hotfix: Swagger format date fix

### LINKED ISSUE #1186: Fix date format in Swagger documentation
## Describe the bug
The schema in Swagger returns an incorrect date format.

## Steps to reproduce the problem
What were you doing when you encountered the problem?
1. Navigate to [Swagger documentation](https://phac-nml.github.io/irida-documentation/swagger-ui/index.html).
2. Look at any schema with a date.
3. Compare it with the actual REST API response.

## Expected behaviour
The Swagger Schema should match the actual REST API response.

## Additional context
Group 1 (12)

### PR DESCRIPTION
**Pull request by ksierks**
## Description of changes
Added schema annotation to specify Long.class returned for createdDate and modifiedDate attributes.

## Related issue
Fixes #1186

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

* [x] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.
~* [ ] Tests added (or description of how to test) for any new features.~
~* [ ] User documentation updated for UI or technical changes.~

### CODE REVIEW

File: src/main/java/ca/corefacility/bioinformatics/irida/model/MutableIridaThing.java [hunk 1]
```diff
@@ -3,34 +3,33 @@
...
 	public void setId(Long id);
-	
 
 	/**
 	 * Get the date that this object was last modified
-	 * 
+	 *
 	 * @return {@link Date} object of the modified date
 	 */
+	@Schema(implementation = Long.class)
```
Comment (ericenns) [inline src/main/java/ca/corefacility/bioinformatics/irida/model/MutableIridaThing.java:26]: Could you add a description here indicating that it is Epoch time e.g.:
```Epoch time in milliseconds when the resource was modified```

File: src/main/java/ca/corefacility/bioinformatics/irida/model/Timestamped.java [hunk 2]
```diff
@@ -11,14 +13,15 @@
 
 	/**
 	 * Get the created date of the object
-	 * 
+	 *
 	 * @return A {@link Date} object of the created date
 	 */
+	@Schema(implementation = Long.class)
```
Comment (ericenns) [inline src/main/java/ca/corefacility/bioinformatics/irida/model/Timestamped.java:19]: Could you add a description here indicating that it is Epoch time e.g.:
```Epoch time in milliseconds when the resource was created```

**Review by ericenns — CHANGES_REQUESTED** [review]
Looks good, I would just like descriptions added so that the API consumer knows that it is epoch formatted.

In addition could you add a schema annotation to `Sample` `collectionDate` e.g.:
```
@Schema(type = "string", format = "date")
```

Also could you please fix issue #1200 in this PR, and also update the version in the pom file.

File: src/main/java/ca/corefacility/bioinformatics/irida/model/MutableIridaThing.java (hunk 1 above)
Comment (ksierks) [reply src/main/java/ca/corefacility/bioinformatics/irida/model/MutableIridaThing.java]: Please see a922b821d63d07ad1000087b7880ef91f815c1a4.

File: src/main/java/ca/corefacility/bioinformatics/irida/model/Timestamped.java (hunk 2 above)
Comment (ksierks) [reply src/main/java/ca/corefacility/bioinformatics/irida/model/Timestamped.java]: Again, please see a922b821d63d07ad1000087b7880ef91f815c1a4.

**Comment by apetkau** [comment]
I wanted to add a note here that the same fix for #1200 (can't save analysis results to sample) added to this hotfix will also fix #1203 (can't concatenate files).

**Comment by ericenns** [comment]
@ksierks could you also add to the changelog that it fixes concatenating files i.e. issue #1203 that @apetkau mentioned.

**Review by ericenns — APPROVED** [review]
👍 Thanks for doing this @ksierks

**Pull request merged**